students/rfelts/lesson03/mailroom.py

```diff
@@ -0,0 +1,142 @@
+        temp_data.append([name_list[index], donation_list[index], len(item[1:]), donation_list[index] / len(item[1:])])
+    name_padding = len(max(name_list, key=len)) + 4
+    donation_padding = len(str(max(donation_list)))
+    donation_padding += int((donation_padding / 3) + 2)
+
+    # Sort the list by donation total (index 1) of the list
+    import operator
+    temp_data.sort(key=operator.itemgetter(1), reverse=True)
+
+    print(f"\n{'Donor Name':<{name_padding}} | {'Total Given':>{donation_padding}} | {'Num Gifts':>03} | " +
+          f"{'Average Gift':<{name_padding}}\n{'-' * 65}")
+    for row in temp_data:
+        print(f"{row[0]:<{name_padding}}  ${row[1]:>{donation_padding},.2f} {row[2]:>11d} {row[3]:>{donation_padding},.2f}")
+
+
+def send_thank_you():
+    """
+    Prompt for a Full Name, to print a list of donors, or to exit.
+    If the user types a name not in the current donor list, add that name to the data structure and use it.
+    Once a name has been selected, prompt for a donation amount.
+    Once an amount has been given, add that amount to the donation history of the selected user.
+    Compose an email thanking the donor for their generous donation.
+    """
+
+    prompt_string = "\nPlease enter the name of the donor, 'list' to see current donors, or 'exit': "
+    name_list = create_name_list()
+    response = input(prompt_string)
+
+    while True:
+        if response.lower() == "exit":
+            break
+        elif response.lower() == "list":
+            print("\n" + ", ".join(name_list))
+            response = input(prompt_string)
+        # Add the new name to the list along with the donation amount then generate the email
+        elif not any(s.lower() == response.lower() for s in name_list):
+            amount = prompt_donation_amount(response)
+            donor_data.append([response, amount])
+            compose_email(response.lower())
+            break
+        # Update the donor with the new donation amount then generate the email
+        elif any(s.lower() == response.lower() for s in name_list):
+            update_donor_history(response.lower(), prompt_donation_amount(response))
+            compose_email(response.lower())
+            break
+
+
+def create_name_list():
+    """
+    Create a list containing the names of the donors
+    :return: a list of donor names
+    """
+    return [item[0] for item in donor_data]
+
+
+def prompt_donation_amount(donor):
+    """
+    Prompt the user for the amount donated
+    :param: string representing the name of the person making the donation
+    :return: a float representing the amount donated
+    """
+    prompt_string = "\nPlease enter " + donor + "'s donation amount: "
+    response = input(prompt_string)
+
+    # Verify the value entered is numeric if not prompt the user for a new value
+    while True:
+        try:
+            response = float(response)
+            return response
+        except:
```

Comment:
It's best to catch the specific exception that you expect to be raised so that you can handle it, `ValueError` in this case